Add a quickstart document with Try Jupyter

willingc opened the pull request:
Addresses #32.

Adds a **Try Jupyter** section for folks that are interested, but perhaps intimidated to install, to try the Notebook.

Adds Quickstart-New User and Quickstart-Experienced User sections with more concise content than the Installation document. Links to the more detailed document.

Revert ``conf.py`` setting for ``html_static_path`` to be consistent with other project repos (like notebook).

Docs landing page:
<img width="902" alt="screen shot 2015-09-21 at 12 48 16 pm" src="https://cloud.githubusercontent.com/assets/2680980/10003033/4e107440-605f-11e5-9dfd-6859049228eb.png">

Quickstart page snippet:
<img width="728" alt="screen shot 2015-09-21 at 12 49 00 pm" src="https://cloud.githubusercontent.com/assets/2680980/10003034/4e11435c-605f-11e5-943a-c9b33de72d99.png">

ellisonbg commented:
I think we definitely want to have a visible top-level link to
try.jupyter.org. But the existing installation documentation on
jupyter.readthedocs.org *is* our quickstart guide to installation and I
don't think that should be repeated here. We have much more detailed
installation information in the individual repos. Also, we have found that
if we have repeated installation instructions in multiple places, they
always fall out of sync.

What about keeping it simple and putting the link to try.jupyter.org at the
very top of the main landing page (or similar).

Also, to me this is a sign that our installation page on
jupyter.readthedocs may be too complex and need to be simplified or
reorganized.

On Mon, Sep 21, 2015 at 12:51 PM, Carol Willing <notifications@github.com>
wrote:

> Addresses #32 <https://github.com/jupyter/jupyter/issues/32>.
>
> Adds a *Try Jupyter* section for folks that are interested, but perhaps
> intimidated to install, to try the Notebook.
>
> Adds Quickstart-New User and Quickstart-Experienced User sections with
> more concise content than the Installation document. Links to the more
> detailed document.
>
> Revert conf.py setting for html_static_path to be consistent with other
> project repos (like notebook).
>
> Docs landing page:
> [image: screen shot 2015-09-21 at 12 48 16 pm]
> <https://cloud.githubusercontent.com/assets/2680980/10003033/4e107440-605f-11e5-9dfd-6859049228eb.png>
>
> Quickstart page snippet:
> [image: screen shot 2015-09-21 at 12 49 00 pm]
> <https://cloud.githubusercontent.com/assets/2680980/10003034/4e11435c-605f-11e5-943a-c9b33de72d99.png>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/jupyter/jupyter/pull/45
> Commit Summary
>
>    - Add quickstart
>    - Add a quickstart guide
>
> File Changes
>
>    - *M* docs/source/conf.py
>    <https://github.com/jupyter/jupyter/pull/45/files#diff-0> (2)
>    - *M* docs/source/index.rst
>    <https://github.com/jupyter/jupyter/pull/45/files#diff-1> (1)
>    - *A* docs/source/quickstart.rst
>    <https://github.com/jupyter/jupyter/pull/45/files#diff-2> (66)
>
> Patch Links:
>
>    - https://github.com/jupyter/jupyter/pull/45.patch
>    - https://github.com/jupyter/jupyter/pull/45.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/jupyter/pull/45>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

ellisonbg commented:
Another option is to replace the Quickstart page with a "Try Jupyter Now"
page that has the first subsection of the existing "Quickstart" page and a
link to the installation page if the user wants to actually start using it.

While we wanted the Installation and Running the Notebook pages to be
top-level, those two pages are really meant to be our quick start guide
(again if they are too complex, let's simplify rather than add).



On Mon, Sep 21, 2015 at 2:04 PM, Brian Granger <ellisonbg@gmail.com> wrote:

> I think we definitely want to have a visible top-level link to
> try.jupyter.org. But the existing installation documentation on
> jupyter.readthedocs.org *is* our quickstart guide to installation and I
> don't think that should be repeated here. We have much more detailed
> installation information in the individual repos. Also, we have found that
> if we have repeated installation instructions in multiple places, they
> always fall out of sync.
>
> What about keeping it simple and putting the link to try.jupyter.org at
> the very top of the main landing page (or similar).
>
> Also, to me this is a sign that our installation page on
> jupyter.readthedocs may be too complex and need to be simplified or
> reorganized.
>
> On Mon, Sep 21, 2015 at 12:51 PM, Carol Willing <notifications@github.com>
> wrote:
>
>> Addresses #32 <https://github.com/jupyter/jupyter/issues/32>.
>>
>> Adds a *Try Jupyter* section for folks that are interested, but perhaps
>> intimidated to install, to try the Notebook.
>>
>> Adds Quickstart-New User and Quickstart-Experienced User sections with
>> more concise content than the Installation document. Links to the more
>> detailed document.
>>
>> Revert conf.py setting for html_static_path to be consistent with other
>> project repos (like notebook).
>>
>> Docs landing page:
>> [image: screen shot 2015-09-21 at 12 48 16 pm]
>> <https://cloud.githubusercontent.com/assets/2680980/10003033/4e107440-605f-11e5-9dfd-6859049228eb.png>
>>
>> Quickstart page snippet:
>> [image: screen shot 2015-09-21 at 12 49 00 pm]
>> <https://cloud.githubusercontent.com/assets/2680980/10003034/4e11435c-605f-11e5-943a-c9b33de72d99.png>
>> ------------------------------
>> You can view, comment on, or merge this pull request online at:
>>
>>   https://github.com/jupyter/jupyter/pull/45
>> Commit Summary
>>
>>    - Add quickstart
>>    - Add a quickstart guide
>>
>> File Changes
>>
>>    - *M* docs/source/conf.py
>>    <https://github.com/jupyter/jupyter/pull/45/files#diff-0> (2)
>>    - *M* docs/source/index.rst
>>    <https://github.com/jupyter/jupyter/pull/45/files#diff-1> (1)
>>    - *A* docs/source/quickstart.rst
>>    <https://github.com/jupyter/jupyter/pull/45/files#diff-2> (66)
>>
>> Patch Links:
>>
>>    - https://github.com/jupyter/jupyter/pull/45.patch
>>    - https://github.com/jupyter/jupyter/pull/45.diff
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/jupyter/jupyter/pull/45>.
>>
>
>
>
> --
> Brian E. Granger
> Associate Professor of Physics and Data Science
> Cal Poly State University, San Luis Obispo
> @ellisonbg on Twitter and GitHub
> bgranger@calpoly.edu and ellisonbg@gmail.com
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

willingc commented:
@ellisonbg Here's another iteration: Added Try Jupyter section. Removed separate quickstart and heavily edited installation doc.

Try Jupyter page:
<img width="993" alt="screen shot 2015-09-21 at 2 42 24 pm" src="https://cloud.githubusercontent.com/assets/2680980/10006007/66ad7cea-606f-11e5-9840-c2873442c910.png">

Edited Installation page:
<img width="1099" alt="screen shot 2015-09-21 at 2 42 56 pm" src="https://cloud.githubusercontent.com/assets/2680980/10006008/66ae6c04-606f-11e5-989f-1bb626750209.png">

minrk commented:
I like the separate try jupyter page

damianavila commented:
> I like the separate try jupyter page

ditto

ellisonbg commented on docs/source/tryjupyter.rst:
```diff
@@ -0,0 +1,16 @@
+Ready to install Jupyter?
+-------------------------
+If you have tried Jupyter and like it, please use our detailed
+:ref:`installation guide <install>` to install it in your
```
I would phrase this as "install it on your computer". I think that "environment" is a bit too vague for new users.

ellisonbg commented on docs/source/install.rst:
```diff
@@ -25,8 +23,8 @@ the IPython kernel.
 
-How to Install Jupyter Notebook
--------------------------------
+Quickstart
```
In my mind "quickstart" or "getting started" includes both installation and running the notebook. I thought about putting that as a heading above those two pages, but I think that is too much hierarchy at this level. Based on this I would just leave it as it was. Also, our previous Google Analytics showed that "install" was the main phrase people were looking for.

ellisonbg commented on docs/source/install.rst:
```diff
@@ -72,8 +70,8 @@ and :ref:`next steps <next-steps>` such as running the Jupyter Notebook.
 
-If you are an existing Python user but new to Jupyter
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Experienced Python user - New to Jupyter
```
I like how you have rephrased these headings. But I would probably use colons like this "Experience Python user: new to Jupyter" on each of them.

ellisonbg commented:
I left a few small points, but I think this looks good overall. Really like the "Try Jupyter" page.

rgbkrk commented:
Awesome. I like the separate page version for Try Jupyter too. Well done.

willingc commented:
Fixed merge conflict.
